src/slack_planning_bot/core/spec.clj

```diff
@@ -0,0 +1,65 @@
+(ns slack-planning-bot.core.spec
+  (require [slack-planning-bot.core.entity :as entity]
+           [clojure.spec.alpha :as s]))
+
+(s/def ::user-id string?)
+(s/def ::message string?)
+
+(s/def ::messenger #(satisfies? entity/Messenger %))
+(s/def ::messenger-error keyword?)
+(s/def ::messenger-result (s/or :data (s/map-of ::user-id ::message)
+                            :error ::messenger-error))
+
+(s/def ::time-tracker #(satisfies? entity/TimeTracker %))
+(s/def ::time-tracker-error keyword?)
+(s/def ::time-tracker-result (s/or :data () :error ::time-tracker-error))
+
+(s/def ::name-provider #(satisfies? entity/MessengerNameProvider %))
+(s/def ::name-provider-error keyword?)
+(s/def ::name-provider-result (s/or :data :user-id :error ::name-provider-error))
+
+(s/def ::scheduler #(satisfies? entity/Scheduler %))
+(s/def ::scheduler-error keyword?)
+(s/def ::scheduler-result (s/or :data () :error ::scheduler-error))
```

Comment:
Thats not right. What is `schedule-job` returning?

Reply:
like above

Reply:
Why won't just check?